### DISCUSSION TITLE
Include a Ultimate GPU switch hotkey for MUX switch problem remedial

### DISCUSSION THREAD

**Discussion by Kowken** [seerge/g-helper#2830] [Ideas]
Greetings.

On some ASUS models equipped with a MUX switch such as mine; an ASUS ROG STRIX SCAR (2022) G533ZW AS94; the MUX switch may fail, ending up with the laptop display being turned off and impossible to turn on (I saw this issue numerous times on Reddit so I can attest that I'm not an isolated case).

The issue is resolved by switching to Ultimate GPU mode (the display works) and back (the display works again). It is however difficult to pintpoint what one is doing if the main display is off and that one doesn't have access to a remote display (I usually take care of this issue by plugging a monitor through the HDMI port to see the GUI to switch to Ultimate GPU mode since the dGPU handles the second monitor).

Thus would it be possible to add a hotkey to switch to Ultimate GPU mode ? Removing the reboot prompt is not really needed since the selector defaults on yes so one can press enter (and it would make one avoid rebooting their computer on accident when they don't have this issue).

Thanks in advance for your answer and have a nice day.
Best regards.

**Comment by seerge** [comment]
@Kowken hi, it's quite exotic case tbh. Won't it be easier to just run a command itself directly w/o any apps ?

Just run this in powershell as admin :) 

Ultimate ON 
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x00090016; Control_status=0}
```

Ultimate OFF
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x00090016; Control_status=01}
```

**Reply by Kowken** [reply]
Hi ! I don't really get how I am supposed to run this command if I can't see anything on my screen or if I make a typo on accident.

**Reply by Kowken** [reply]
I also get a generic error when running this command:

Invoke-CimMethod : Échec générique
Au caractère Ligne:1 : 1
+ Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName Asus ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (AsusAtkWmi_WMNB...PNP0C14\ATK_0"):CimInstance) [Invoke-CimMethod], CimExc
   eption
    + FullyQualifiedErrorId : HRESULT 0x80041001,Microsoft.Management.Infrastructure.CimCmdlets.InvokeCimMethodCommand

**Reply by Kowken** [reply]
I saw there were shortcuts to enable either Eco or Standard mode but not Ultimate, hence my question.
Otherwise, is there a way to configure a custom hotkey for this case ? I don't really get how the latters work in this app.

**Reply by seerge** [reply]
@Kowken current Eco / Standard hotkeys are absolutely intentionally done in such a way that you don't accidentally hit them (``Ctrl + Shift + Alt + F14``). They are designed for letting people trigger that action from other app (like Autohotkey). How would you perform something like that w/o minitor for Ultimate switch ? :)

Btw, on Scar 16 2023 that I had for some time - it was absolutely possible to go into in BIOS (F2 on boot) and then swithc there between set "dynamic GPU" (or something like that) setting there ON or OFF. It was Ultimate mode switch . You should have that as well.

**Reply by Kowken** [reply]
> @Kowken current Eco / Standard hotkeys are absolutely intentionally done in such a way that you don't accidentally hit them (``Ctrl + Shift + Alt + F14``). They are designed for letting people trigger that action from other app (like Autohotkey). How would you perform something like that w/o minitor for Ultimate switch ? :)
> 
> 
> 
> Btw, on Scar 16 2023 that I had for some time - it was absolutely possible to go into in BIOS (F2 on boot) and then swithc there between set "dynamic GPU" (or something like that) setting there ON or OFF. It was Ultimate mode switch . You should have that as well.

I just checked in the BIOS and there is no such thing on mine (I made sure my BIOS was up to date so it may be starting 2023 models).

To answer your question, you don't really need a monitor if you issue the same kind of command that when you click on ultimate GPU mode. The window asking you to reboot or not has its selector enabled meaning that you can use the directional keys and then press enter. Plus, the default choice you end up on using the selector is yes so you would just have to perform the shortcut and then press enter.

**Reply by Kowken** [reply]
Also is the BIOS menu even displayed when the MUX switch fails ? The boot logo doesn't show at all but I never tried the BIOS (though I suppose it's the same)

**Reply by seerge** [reply]
@Kowken i think BIOS menu works anyway. You can also pressing and holding power button for 30-40 seconds, that usually resets all settings including MUX

**Reply by Kowken** [reply]
BIOS menu does not contain the graphic settings you mentioned on models prior to 2023 though.
Is it possible to leave this thread open until this error happens again so I can test that and the 40 seconds reset ?
Thanks in advance

**Reply by seerge** [reply]
@Kowken sure :)

How did you trigger that ? I once managed to trigger that on Strix by setting Ultimate directly from Eco.

**Reply by Kowken** [reply]
> @Kowken sure :)
> 
> 
> 
> How did you trigger that ? I once managed to trigger that on Strix by setting Ultimate directly from Eco.

Thank you so much !
It mostly happened when I put my computer on hibernate or close the lid but it is quite (fortunately) inconsistent.
I will test the Eco to Ultimate transition.
